src/client/views/components/todoItem.jsx

```diff
@@ -75,12 +75,16 @@ export default React.createClass({
     },
 
     handleBack: function () {
-        const {description, dueDate, title} = this.props;
-        this.setState({
-            inEdit: false,
-            description,
-            dueDate: dueDate.toDateString(),
-            title});
+        const {description, dueDate, title, addingNewItem} = this.props;
+        if (addingNewItem) {
+            this.handleDelete();
+        } else {
+            this.setState({
+                inEdit: false,
+                description,
+                dueDate: dueDate.toDateString(),
+                title});
+        }
```

Comment:
This is good. Another way you could approach this is:
```
if (addingNewItem) { 
    return this.handleDelete();
}
this.setState({ ... });
```
Then you don't need the else, and it keeps the code a little shorter. But it's also a little less _explicit_.